web/ui/react-app/src/pages/serviceDiscovery/Services.tsx

```diff
@@ -82,14 +91,41 @@ export const processTargets = (activeTargets: Target[], droppedTargets: DroppedT
 };
 
 export const ServiceDiscoveryContent: FC<ServiceMap> = ({ activeTargets, droppedTargets }) => {
-  const targets = processSummary(activeTargets, droppedTargets);
-  const labels = processTargets(activeTargets, droppedTargets);
+  const [activeTargetList, setActiveTargetList] = useState(activeTargets);
+  const [targetList, setTargetList] = useState(processSummary(activeTargets, droppedTargets));
+  const [labelList, setLabelList] = useState(processTargets(activeTargets, droppedTargets));
+
+  const handleSearchChange = (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    if (e.target.value !== '') {
+      const result = kvSearch.filter(e.target.value.trim(), activeTargets);
+      setActiveTargetList(
+        result.map((value) => {
+          return value.original as unknown as Target;
+        })
+      );
+    } else {
+      setActiveTargetList(activeTargets);
+    }
+  };
+
+  useEffect(() => {
+    setTargetList(processSummary(activeTargetList, droppedTargets));
+    setLabelList(processTargets(activeTargetList, droppedTargets));
+  }, [activeTargetList, droppedTargets]);
 
   return (
     <>
       <h2>Service Discovery</h2>
+      <Container>
+        <InputGroup>
+          <InputGroupAddon addonType="prepend">
+            <InputGroupText>{<FontAwesomeIcon icon={faSearch} />}</InputGroupText>
+          </InputGroupAddon>
+          <Input autoFocus onChange={handleSearchChange} placeholder="Filter by endpoint or labels" />
```

Comment:
The placeholder text should be adjusted for the service discovery page, since there's no endpoint column.

Reply:
fixed.